pkg/cluster/cache.go

```diff
@@ -0,0 +1,57 @@
+package cluster
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+	"sync"
+)
+
+var clusterCache = kubeFedClusterClients{clusters: map[string]*FedCluster{}}
+
+type kubeFedClusterClients struct {
+	sync.Mutex
+	clusters map[string]*FedCluster
+}
+
+// ClusterData stores cluster client and previous health check probe results of individual cluster.
+type FedCluster struct {
+	// clusterKubeClient is the kube client for the cluster.
+	Client client.Client
+
+	Name              string
+	Type              Type
+	OperatorNamespace string
+
+	// clusterStatus is the cluster status as of last sampling.
+	ClusterStatus *v1beta1.KubeFedClusterStatus
+}
+
+func (c kubeFedClusterClients) addFedCluster(cluster *FedCluster) {
```

Comment:
That was on my todo list as well :-) - I have already some local changes, including this and some more so I won't commit your suggestion as it would introduce conflicts on my side